go/cmd/vtctldclient/internal/command/keyspaces.go

```diff
@@ -48,8 +66,119 @@ var (
+	// RemoveKeyspaceCell makes a RemoveKeyspaceCell gRPC call to a vtctld.
+	RemoveKeyspaceCell = &cobra.Command{
+		Use:  "RemoveKeyspaceCell <keyspace> <cell>",
+		Args: cobra.ExactArgs(2),
+		RunE: commandRemoveKeyspaceCell,
+	}
 )
 
+var createKeyspaceOptions = struct {
+	Force             bool
+	AllowEmptyVSchema bool
+
+	ShardingColumnName string
+	ShardingColumnType cli.KeyspaceIDTypeFlag
+
+	ServedFromsMap cli.StringMapValue
+
+	KeyspaceType      cli.KeyspaceTypeFlag
+	BaseKeyspace      string
+	SnapshotTimestamp string
+}{
+	KeyspaceType: cli.KeyspaceTypeFlag(topodatapb.KeyspaceType_NORMAL),
+}
+
+func commandCreateKeyspace(cmd *cobra.Command, args []string) error {
+	name := cmd.Flags().Arg(0)
+
+	switch topodatapb.KeyspaceType(createKeyspaceOptions.KeyspaceType) {
+	case topodatapb.KeyspaceType_NORMAL, topodatapb.KeyspaceType_SNAPSHOT:
+	default:
+		return errors.New("invalid keyspace type passed to --type")
+	}
+
+	var snapshotTime *vttime.Time
+	if topodatapb.KeyspaceType(createKeyspaceOptions.KeyspaceType) == topodatapb.KeyspaceType_SNAPSHOT {
+		if createKeyspaceOptions.BaseKeyspace == "" {
+			return errors.New("--base-keyspace is required for a snapshot keyspace")
+		}
+
+		if createKeyspaceOptions.SnapshotTimestamp == "" {
+			return errors.New("--snapshot-timestamp is required for a snapshot keyspace")
+		}
+
+		t, err := time.Parse(time.RFC3339, createKeyspaceOptions.SnapshotTimestamp)
+		if err != nil {
+			return err
+		}
+
+		if now := time.Now(); t.After(now) {
+			return fmt.Errorf("--snapshot-time cannot be in the future; snapshot = %v, now = %v", t, now)
+		}
+
+		snapshotTime = logutil.TimeToProto(t)
+	}
+
+	req := &vtctldatapb.CreateKeyspaceRequest{
+		Name:               name,
+		Force:              createKeyspaceOptions.Force,
+		AllowEmptyVSchema:  createKeyspaceOptions.AllowEmptyVSchema,
+		ShardingColumnName: createKeyspaceOptions.ShardingColumnName,
+		ShardingColumnType: topodatapb.KeyspaceIdType(createKeyspaceOptions.ShardingColumnType),
+		Type:               topodatapb.KeyspaceType(createKeyspaceOptions.KeyspaceType),
+		BaseKeyspace:       createKeyspaceOptions.BaseKeyspace,
+		SnapshotTime:       snapshotTime,
+	}
+
+	for n, v := range createKeyspaceOptions.ServedFromsMap.StringMapValue {
+		tt, err := topo.ParseServingTabletType(n)
+		if err != nil {
+			return err
+		}
+
+		req.ServedFroms = append(req.ServedFroms, &topodatapb.Keyspace_ServedFrom{
+			TabletType: tt,
+			Keyspace:   v,
+		})
+	}
+
+	resp, err := client.CreateKeyspace(commandCtx, req)
+	if err != nil {
+		return err
+	}
+
+	data, err := cli.MarshalJSON(resp.Keyspace)
+	if err != nil {
+		return err
+	}
+
+	fmt.Printf("Successfully created keyspace %s. Result:\n%s\n", name, data)
+
+	return nil
+}
+
+var deleteKeyspaceOptions = struct {
+	Recursive bool
+}{}
+
+func commandDeleteKeyspace(cmd *cobra.Command, args []string) error {
+	ks := cmd.Flags().Arg(0)
+
+	_, err := client.DeleteKeyspace(commandCtx, &vtctldatapb.DeleteKeyspaceRequest{
```

Comment:
That's a great point. I started down the following compromise:

```go
resp, err := client.DeleteKeyspace(...)
if err != nil {
    return fmt.Errorf("...", ks, err)
}

defer func() { fmt.Fprintf(os.Stderr, "Successfully deleted keyspace %v.\n", ks }()

data, err := cli.MarshalJSON(resp)
if err != nil {
    return err
}

fmt.Printf("%s\n", data)

return nil
```

Which produces:

```
❯ vtctldclient --server "localhost:15999" DeleteKeyspace -r commerce
{

}
Successfully deleted keyspace commerce.
```

And I _kinda_ wish I got back bytes that let me do something like `if len(data) > 0 { fmt.Printf("%s\n", data) }` but I'm also not sure how to do that without losing the pretty printing (and also the curly braces for the empty-but-non-null object).

What we _could_ do now that I'm thinking this through is expand `cli.MarshalJSON`'s `proto.Message` case to `protoreflect` on the message and return an empty slice if there are no real fields in the message. However. That only works if we're using `google.golang.org/protobuf/proto`, which provides a `protoreflect` package that works with the `proto.Message` type defined in that module, but does not work with the types in `github.com/golang/protobuf/proto`, because the former is technically a v2 of the latter and they didn't keep backwards compatibility on this one.

So I think either of these has to be Good Enough, but I'm also curious if you have other ideas after reading all this:
1. The "Really Hope We Don't Forget" approach. We could also add something to the PR template, but that feels a bit heavy-handed since it will not apply to something like 99% of PRs.
2. Accept that these commands will print an empty object over 3 lines (`[]byte("{\n\n}")`, plus the newline from `fmt.Printf("%s\n", data)`)